File: CegCRMAPI/CegCRMAPI/Core/CegCRMAPI.Domain/Entities/Sale.cs
using CegCRMAPI.Domain.Entities.Common;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CegCRMAPI.Domain.Entities
{
    public class Sale : BaseEntity
    {
        public DateTime SaleDate { get; set; }
        public Guid CustomerId { get; set; }
        public Guid SalesPersonId { get; set; }
        public decimal TotalAmount { get; set; }
        public decimal Discount { get; set; }
        public decimal Tax { get; set; }
        public decimal FinalAmount { get; set; }
        public SaleStatus Status { get; set; } 
        public string InvoiceNumber { get; set; } = string.Empty;

        // Navigation properties
        public Customer Customer { get; set; } = null!;
        public Employee SalesPerson { get; set; } = null!;
        public ICollection<SaleProduct> SaleProducts { get; set; } = new List<SaleProduct>();

        public enum SaleStatus
        {
            [Display(Name = "Proposal")]
            Proposal = 1,

            [Display(Name = "Negotiation")]
            Negotiation = 2,

            [Display(Name = "Accepted")]
            Accepted = 3,

            [Display(Name = "Rejected")]
            Rejected = 4,

            [Display(Name = "Completed")]
            Completed = 5
        }
    }
}

File: CegCRMAPI/CegCRMClient/src/types/sale.ts
export enum SaleStatus {
  Pending = 0,
  Completed = 1,
  Cancelled = 2
}

export interface Sale {
  id: string;
  saleDate: string;
  customerId: string;
  salesPersonId: string;
  totalAmount: number;
  discount: number;
  tax: number;
  finalAmount: number;
  status: SaleStatus; //enum
  invoiceNumber: string | null;
  createdDate: string;
  updatedDate: string | null;
  saleProducts: SaleProduct[] | null;
}

export interface CreateSale {
  saleDate: string;
  customerId: string;
  salesPersonId: string;
  totalAmount: number;
  discount: number;
  tax: number;
  finalAmount: number;
  status: string | null;
  invoiceNumber: string | null;
  products: SaleProductItem[] | null;
}

export interface SaleProduct {
  productId: string;
  productName: string | null;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
}

export interface SaleProductItem {
  productId: string;
  quantity: number;
  unitPrice: number;
} 